app/src/main/java/com/marvel/marvel/main/model/QueryMap.kt

```diff
@@ -0,0 +1,56 @@
+package com.marvel.marvel.main.model
+
+
+import android.util.Log
+import java.security.MessageDigest
+import java.security.NoSuchAlgorithmException
+import java.util.*
+
+internal class QueryMap : LinkedHashMap<String, String>() {
+
+    companion object {
+        private const val PUBLIC_KEY = "c735e50e07fb1d2384b908dea2ef423a" //TODO inject and move to environment variable
```

Comment:
I agree with the comment. Check out how it is done here:
https://github.com/MarcinMoskala/MarvelGallery